tasks.py

```diff
@@ -97,19 +153,31 @@ def destroy(context):
     docker_compose(context, "down --volumes")
 
 
-@task
-def vscode(context):
+@task(help={"python-ver": f"The version of Python to build the container with (default: {DEFAULT_PYTHON_VER})"})
+def vscode(context, python_ver=None):
     """Launch Visual Studio Code with the appropriate Environment variables to run in a container."""
-    context.run("code nautobot.code-workspace", env={"PYTHON_VER": PYTHON_VER})
+    command = "code nautobot.code-workspace"
+
+    if python_ver is None:
+        python_ver = context.nautobot.python_ver
+
+    context.run(command, env={"PYTHON_VER": python_ver})
 
 
 # ------------------------------------------------------------------------------
 # ACTIONS
 # ------------------------------------------------------------------------------
-@task
-def nbshell(context):
+@task(help={"local": "run this task locally vs inside the docker container (default: False)"})
+def nbshell(context, local=None):
```

Comment:
Same comment/concern about adding `local` as a CLI parameter to a bunch of tasks.

Reply:
I'm starting to lean more towards pulling out the `--python-ver`, however, right now AFIK we don't have a defined/documented way of running postgres and redis locally so in my poetry dev env for example I can easily run all of the linters locally but tests which require a DB I run inside the docker container

Reply:
I'm actually kind of interested in the idea of having a compose override to only spin postgres/redis so I can run all of the tests locally.

Reply:
> I'm starting to lean more towards pulling out the `--python-ver`, however, right now AFIK we don't have a defined/documented way of running postgres and redis locally so in my poetry dev env for example I can easily run all of the linters locally but tests which require a DB I run inside the docker container

This isn't entirely true, because we point to it in the docs. Either install PG/Redis locally yourself, or use them in containers. We haven't documented a hybrid approach of, say, using containers for PG/Redis and running Nautobot locally from a virtualenv. 

In any case, doing it all by way of the Invoke config or environment variables seems like the best path IMO, since we want `local` to be a global boolean, don't we? I think we just need to stick with one path, because most people aren't mixing/matching their development environments. Let's not try to boil the ocean just yet!